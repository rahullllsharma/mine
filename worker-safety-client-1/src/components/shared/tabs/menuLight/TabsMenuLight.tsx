import type { ItemClasses, TabsOption, TabsProps } from "../Tabs";
import Tabs from "../Tabs";

export default function TabsMenuLight({
  options,
  defaultIndex,
  selectedIndex,
  onSelect,
}: Omit<TabsProps, "itemClasses" | "options"> & {
  options: Pick<TabsOption, "id" | "name">[] | string[];
  selectedIndex?: number;
}): JSX.Element {
  const itemClasses: ItemClasses = {
    default: "text-sm text-neutral-shade-100 box-border",
    selected:
      "border-t-2 border-solid border-brand-urbint-40 text-neutral-shade-100",
    unselected: "text-neutral-shade-58",
  };

  return (
    <>
      <Tabs
        itemClasses={itemClasses}
        options={options}
        defaultIndex={defaultIndex}
        selectedIndex={selectedIndex}
        onSelect={onSelect}
        type="menu"
      />
    </>
  );
}
