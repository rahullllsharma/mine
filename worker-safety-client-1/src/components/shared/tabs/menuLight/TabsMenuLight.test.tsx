import { render } from "@testing-library/react";
import TabsMenuLight from "./TabsMenuLight";

const options = ["Active", "Pending", "Completed"];

describe(TabsMenuLight.name, () => {
  it("should match the snapshot", () => {
    const { asFragment } = render(
      <TabsMenuLight options={options} onSelect={jest.fn()} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
