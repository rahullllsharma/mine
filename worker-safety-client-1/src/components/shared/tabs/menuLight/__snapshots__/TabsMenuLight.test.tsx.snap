// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsMenuLight should match the snapshot 1`] = `
<DocumentFragment>
  <div
    aria-orientation="horizontal"
    class="flex h-full w-full"
    role="tablist"
  >
    <button
      aria-selected="true"
      class="min-w-0 z-10 outline-none w-1/3"
      id="headlessui-tabs-tab-1"
      role="tab"
      tabindex="0"
      type="button"
    >
      <div
        class="flex items-center flex-col h-full gap-0 pt-2 text-sm text-neutral-shade-100 box-border border-t-2 border-solid border-brand-urbint-40 text-neutral-shade-100"
      >
        <span
          class="text-sm truncate"
        >
          Active
        </span>
      </div>
    </button>
    <button
      aria-selected="false"
      class="min-w-0 z-10 outline-none w-1/3"
      id="headlessui-tabs-tab-2"
      role="tab"
      tabindex="-1"
      type="button"
    >
      <div
        class="flex items-center flex-col h-full gap-0 pt-2 text-sm text-neutral-shade-100 box-border text-neutral-shade-58"
      >
        <span
          class="text-sm truncate"
        >
          Pending
        </span>
      </div>
    </button>
    <button
      aria-selected="false"
      class="min-w-0 z-10 outline-none w-1/3"
      id="headlessui-tabs-tab-3"
      role="tab"
      tabindex="-1"
      type="button"
    >
      <div
        class="flex items-center flex-col h-full gap-0 pt-2 text-sm text-neutral-shade-100 box-border text-neutral-shade-58"
      >
        <span
          class="text-sm truncate"
        >
          Completed
        </span>
      </div>
    </button>
  </div>
</DocumentFragment>
`;
